packages/react-components/src/components/VideoGallery/TogetherModeStream.tsx

```diff
@@ -0,0 +1,120 @@
+/* @conditional-compile-remove(together-mode) */
+import { Stack } from '@fluentui/react';
+/* @conditional-compile-remove(together-mode) */
+import { togetherModeStreamRootStyle } from '../styles/TogetherMode.styles';
+/* @conditional-compile-remove(together-mode) */
+/**
+ * A memoized version of local screen share component. React.memo is used for a performance
+ * boost by memoizing the same rendered component to avoid rerendering this when the parent component rerenders.
+ * https://reactjs.org/docs/react-api.html#reactmemo
+ */
+export const TogetherModeStream = React.memo(
+  (props: {
+    startTogetherModeEnabled?: boolean;
+    isTogetherModeActive?: boolean;
+    onCreateTogetherModeStreamView?: (options?: VideoStreamOptions) => Promise<void | TogetherModeStreamViewResult>;
+    onStartTogetherMode?: (options?: VideoStreamOptions) => Promise<void | TogetherModeStreamViewResult>;
+    onDisposeTogetherModeStreamView?: () => Promise<void>;
+    onSetTogetherModeSceneSize?: (width: number, height: number) => void;
+    togetherModeStreams?: VideoGalleryTogetherModeStreams;
+    seatingCoordinates?: VideoGalleryTogetherModeParticipantPosition;
+    reactionResources?: ReactionResources;
+    localParticipant?: VideoGalleryLocalParticipant;
+    remoteParticipants?: VideoGalleryRemoteParticipant[];
+    screenShareComponent?: JSX.Element;
+    containerWidth?: number;
+    containerHeight?: number;
+  }): React.ReactNode => {
+    const {
+      startTogetherModeEnabled,
+      isTogetherModeActive,
+      onCreateTogetherModeStreamView,
+      onStartTogetherMode,
+      onSetTogetherModeSceneSize,
+      onDisposeTogetherModeStreamView,
+      togetherModeStreams,
+      containerWidth,
+      containerHeight
+    } = props;
+
+    useEffect(() => {
+      return () => {
+        // TODO: Isolate disposing behaviors for screenShare and videoStream
+        onDisposeTogetherModeStreamView && onDisposeTogetherModeStreamView();
+      };
+    }, [onDisposeTogetherModeStreamView]);
+
+    // Trigger startTogetherMode only when needed
+    useEffect(() => {
+      if (startTogetherModeEnabled && !isTogetherModeActive) {
+        onStartTogetherMode?.();
+      }
+    }, [startTogetherModeEnabled, isTogetherModeActive, onStartTogetherMode]);
+
+    // Create stream view if not already created
+    useEffect(() => {
+      if (!togetherModeStreams?.mainVideoStream?.renderElement) {
+        onCreateTogetherModeStreamView?.();
+      }
+    }, [togetherModeStreams?.mainVideoStream?.renderElement, onCreateTogetherModeStreamView]);
+
+    // Update scene size only when container dimensions change
+    const reCalculateSeatingPosition = useMemo(() => {
+      if (onSetTogetherModeSceneSize && containerWidth && containerHeight) {
+        onSetTogetherModeSceneSize(containerWidth, containerHeight);
+      }
+    }, [onSetTogetherModeSceneSize, containerWidth, containerHeight]);
+
+    useEffect(() => {
```

Comment:
we want to avoid having a hook with no content. if we are using hooks like this then we should re-evaluate how we are handling the props and the container size